seedfarmer/mgmt/archive_support.py

```diff
@@ -72,13 +72,33 @@ def _download_archive(archive_url: str, secret_name: Optional[str]) -> Response:
 def _extract_archive(archive_name: str, extracted_dir_path: str) -> str:
     if archive_name.endswith(".tar.gz"):
         with tarfile.open(archive_name, "r:gz") as tar_file:
-            embedded_dir = os.path.commonprefix(tar_file.getnames())
-            tar_file.extractall(extracted_dir_path)
+            all_members = tar_file.getmembers()
+            top_level_dirs = set(member.name.split("/")[0] for member in all_members if "/" in member.name)
+            if len(top_level_dirs) > 1:
+                raise InvalidConfigurationError(
+                    f"the archive {archive_name} can only have one directory at the root and no files"
+                )
+            elif len(top_level_dirs) == 1:
+                embedded_dir = top_level_dirs.pop()
+            else:
+                embedded_dir = ""
+            members_to_extract = [m for m in all_members]
```

Comment:
What's the goal of this? Do you have to make a copy for some reason? (Similar question for the zip archive).

Reply:
This is to move the dir that represents the module to a relative path that matches the `modules` query parameter as if it were in a project.   The initial design of SF was to have a module in a location with a relative path, so the references are relative to seedfarmer.yaml (the project reference).  This move aligns the modules in the archive to that paradigm.

Reply:
I think I was unclear -- I'm specifically talking about the line `members_to_extract = [m for m in all_members]`. `members_to_extract` is identical to `all_members` as far as I can tell?

Reply:
I removed the generator